### PR TITLE
feat(hashs): compute object hashes by chunk data

### PR DESCRIPTION
**Pull request by giangbui**
Compute hashes by progressively updating hashes with chunk data

### CODE REVIEW

File: handlers/s3client.go [hunk 1]
```diff
@@ -48,3 +51,28 @@ func StreamObjectFromS3(client *AwsClient, bucket string, key string) ([]byte, e
...
+// GetObjectSize returns object size in bytes
+func GetObjectSize(client *AwsClient, bucket string, key string) (*int64, error) {
+	svc := s3.New(client.session)
+	input := &s3.HeadObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	}
+
+	result, err := svc.HeadObject(input)
+	if err != nil {
```
Comment (rudyardrichter) [inline handlers/s3client.go]: I'm confused what this block is doing, couldn't this just be
```go
fmt.Println(err.Error())
return nil, err
```
or does casting it to `awserr.Error` actually return a different message?

File: handlers/s3client.go [hunk 2]
```diff
@@ -20,7 +22,7 @@ func CreateNewAwsClient() (*AwsClient, error) {
 	client := new(AwsClient)
 
 	sess, err := session.NewSession(&aws.Config{
-		Region: aws.String(os.Getenv("AWS_REGION")),
+		Region: aws.String("us-east-1"),
```
Comment (rudyardrichter) [inline handlers/s3client.go]: try to get AWS_REGION from environment and use this for default?

File: handlers/s3client.go [hunk 3]
```diff
@@ -33,13 +35,14 @@ func CreateNewAwsClient() (*AwsClient, error) {
 	return client, nil
 }
 
-// StreamObjectFromS3 downloads object from s3
-func StreamObjectFromS3(client *AwsClient, bucket string, key string) ([]byte, error) {
+// GetChunkDataFromS3 downloads chuck data from s3
```
Comment (rudyardrichter) [inline handlers/s3client.go]: typo: ~`chuck`~ -> `chunk`

File: handlers/s3client.go [hunk 4]
```diff
@@ -48,3 +51,28 @@ func StreamObjectFromS3(client *AwsClient, bucket string, key string) ([]byte, e
 	return buff.Bytes(), nil
 
 }
+
+// GetObjectSize returns object size in bytes
+func GetObjectSize(client *AwsClient, bucket string, key string) (*int64, error) {
```
Comment (rudyardrichter) [inline handlers/s3client.go]: what is the point of the `*int64` type? does the `nil` value represent the object doesn't exist?

File: handlers/s3client.go (hunk 1 above)
Comment (giangbui) [reply handlers/s3client.go]: Errors returned by the SDK are backed by a concrete type that will satisfy the awserr.Error interface.

https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/handling-errors.html

File: handlers/s3client.go (hunk 3 above)
Comment (giangbui) [reply handlers/s3client.go]: Fixed in f6d90ae451c37d43f7ee3f6e7fd753ccdb1f6580

File: handlers/s3client.go (hunk 4 above)
Comment (giangbui) [reply handlers/s3client.go]: I am using AWS s3 HeadObject object containing a pointer to the size of an object. If the object does not exist, the pointer is nil

File: handlers/s3client.go (hunk 2 above)
Comment (giangbui) [reply handlers/s3client.go]: Switch back to env

**Pull request merged**